components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -374,29 +371,26 @@ func (g *rtkSerial) connectToNTRIP() error {
 	}
 	err := g.connectAndParseSourceTable()
 	if err != nil {
-		return g.err.Get()
+		return err
 	}
 
 	err = g.openPort()
 	if err != nil {
-		g.err.Set(err)
-		return g.err.Get()
```

Comment:
Note: of course the old version on line 377 could return a nil error even when `err` was non-nil, but this part was subtly wrong, too: if we called `g.err.Set(err)`, then some other goroutine called `g.err.Get()`, and then we tried calling `g.err.Get()` afterwards, we could still return nil after encountering a non-nil error, leading to further trouble.